### ISSUE TITLE
Opam package calls the 'install' tool with an illegal option on OSX

### ISSUE THREAD

**Issue by beurdouche** [FStarLang/FStar#469]
When installing the fstar opam package on OSX `$ opam install fstar`
there is yet another boring incompatibility between Linux and Unix tools...
```
# install -m 755 -D -T fstar.exe /Users/bbeurdouche/.opam/system/bin/fstar.exe
### stderr ###
# [...]
# Warning 26: unused variable for1_sig.
# File "FStar_Extraction_ML_ExtractExp.ml", line 547, characters 8-10:
# Warning 26: unused variable t0.
# install: illegal option -- D
# usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
#                [-o owner] file1 file2
#        install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
#                [-o owner] file1 ... fileN directory
#        install -d [-v] [-g group] [-m mode] [-o owner] directory ...
# make: *** [install] Error 64
```

**Comment by msprotz** [comment]
Urgh. Any ideas Benjamin?

**Comment by beurdouche** [comment]
Unfortunately, I am not sure there is a generic way to circumvent this kind of problems...

That said, everyone that has installed `coreutils` through Homebrew or Macports has all those utilities in the GNU version available by prefixing the tools with `g` (`ginstall -D` will work properly while `install -D` fails).

A fix could be to simply search if `ginstall` exists and use it if it is available.
It's not generic, still it will be a good fix a lot of people...

**Comment by beurdouche** [comment]
@msprotz, I think, I'll patch the Makefile this way in order to fix (#468) ...

**Comment by msprotz** [comment]
Yes that would be good.

That sounds like one more argument in favor of a configure... see the last few lines of https://github.com/protz/mezzo/blob/master/configure to see how to detect this kind of stuff

**Comment by beurdouche** [comment]
Fixed by 85f3ba4e
